### PR TITLE
Adding scala profile for kafka 2.x build and remove root pom scala dependencies

### PR DESCRIPTION
**Pull request by xiangfu0**
## Description
Adding scala profile for kafka 2.x build and remove root pom scala dependencies

## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
* [ ] Yes (Please label as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR fix a zero-downtime upgrade introduced earlier?
* [ ] Yes (Please label this as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR otherwise need attention when creating release notes? Things to consider:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed
* [ ] Yes (Please label this PR as **<code>release-notes</code>** and complete the section on Release Notes)

### CODE REVIEW

File: pinot-plugins/pinot-stream-ingestion/pinot-kafka-2.0/pom.xml [hunk 1]
```diff
@@ -100,4 +98,28 @@
       <artifactId>jopt-simple</artifactId>
     </dependency>
   </dependencies>
+
+
+  <profiles>
+    <profile>
+      <id>scala-2.12</id>
+      <activation>
+        <activeByDefault>true</activeByDefault>
```
Comment (jackjlli) [inline pinot-plugins/pinot-stream-ingestion/pinot-kafka-2.0/pom.xml:107]: This basically means the default scala version got downgraded from 2.13 to 2.12 in the project. Could you elaborate a bit more on the reason of doing that to the description?

Comment (xiangfu0) [reply pinot-plugins/pinot-stream-ingestion/pinot-kafka-2.0/pom.xml:107]: The reason is I've got a lot request for scala 2.12 support, hence make this as default and allow upgrade to 2.13.

Basically this is similar to kafka 0.9 and 2.x, users can build a connector based on the scala version from their enviroment.

Comment (xiangfu0) [reply pinot-plugins/pinot-stream-ingestion/pinot-kafka-2.0/pom.xml:107]: For Pinot users, there is no different in terms of using scala 2.12 or 2.13 built jars.
This is mostly for users who are integrating Pinot code and libs and making their life easier.

**Review by Jackie-Jiang — APPROVED** [review]

**Review by jackjlli — APPROVED** [review]
LGTM

**Pull request merged**